controllers/namespacescope/namespacescope.go

```diff
@@ -0,0 +1,117 @@
+//
+// Copyright 2021 IBM Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package nss
+
+import (
+	"context"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+	utilwait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/klog"
+
+	nssv1 "github.com/IBM/ibm-namespace-scope-operator/api/v1"
+
+	"github.com/IBM/ibm-common-service-operator/controllers/bootstrap"
+)
+
+var (
+	apiGroupVersion = "operator.ibm.com/v1"
+	Kinds           = []string{"NamespaceScope"}
+	CRList          = []string{"common-service", "nss-odlm-scope"}
+	sourceCR        = "common-service"
+	targetCR        = "nss-odlm-scope"
+)
+
+var ctx = context.Background()
+
+// SyncUpCR syncs up the namespace members in source CR and target CR
+func SyncUpCR(bs *bootstrap.Bootstrap) {
+	for {
+		// wait for nss CRD
+		for _, kind := range Kinds {
+			if err := bs.WaitResourceReady(apiGroupVersion, kind); err != nil {
+				klog.Errorf("Failed to wait for resource ready with kind %s, apiGroupVersion: %s", kind, apiGroupVersion)
+				continue
+			}
+		}
+
+		// wait for source and target CR
+		for _, cr := range CRList {
+			for {
+				ready := waitCRReady(bs, cr, bs.CSData.MasterNs)
+				if ready {
+					break
+				}
+				time.Sleep(10 * time.Second)
+			}
+		}
+
+		// fetch the source and target NSS CR
+		sourceNsScope := &nssv1.NamespaceScope{}
+		sourceNsScopeKey := types.NamespacedName{Name: sourceCR, Namespace: bs.CSData.MasterNs}
+		if err := bs.Reader.Get(ctx, sourceNsScopeKey, sourceNsScope); err != nil {
+			klog.Errorf("Failed to get NSS CR %s: %v, retry again", sourceNsScopeKey.String(), err)
+			continue
+		}
+
+		targetNsScope := &nssv1.NamespaceScope{}
+		targetNsScopeKey := types.NamespacedName{Name: targetCR, Namespace: bs.CSData.MasterNs}
+		if err := bs.Reader.Get(ctx, targetNsScopeKey, targetNsScope); err != nil {
+			klog.Errorf("Failed to get NSS CR %s: %v, retry again", targetNsScopeKey.String(), err)
+			continue
+		}
+
+		// sync up when namepsace in source CR is different from target CR
+		if len(sourceNsScope.Spec.NamespaceMembers) != len(targetNsScope.Spec.NamespaceMembers) {
```

Comment:
Using the length to check the namespace item is not very accurate.

Could we leverage "github.com/deckarep/golang-set"  to do the comparison?

It has a function ["Difference"](https://pkg.go.dev/github.com/deckarep/golang-set@v1.7.1#Set.Difference) to get the difference of two sets.

Reply:
It sounds good, by using the set, we will have a precise comparison.

Reply:
@horis233 @ZhuoxiLi 

The PR is ready for reviewing.

In order to keep the consistency of two NSS CRs, I choose to overwrite the entire `nss-odlm-scope` namespace members by `common-service` members once they are different.

In this case, the changes in `nss-odlm-scope` CR will not take effect and the `common-service` CR is the only one which will be changed by users.

Reply:
Will the overwriting the entire nss-odlm-scope cause they are always the same? It will impact the multiple instances installation, right?